cmd/src/debug_kube.go

```diff
@@ -0,0 +1,373 @@
+package main
+
+import (
+	"archive/zip"
+	"bytes"
+	"context"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"golang.org/x/sync/errgroup"
+
+	"golang.org/x/sync/semaphore"
+
+	"github.com/sourcegraph/src-cli/internal/cmderrors"
+	"github.com/sourcegraph/src-cli/internal/exec"
+)
+
+func init() {
+	usage := `
+'src debug kube' mocks kubectl commands to gather information about a kubernetes sourcegraph instance. 
+
+Usage:
+
+    src debug kube [command options]
+
+Flags:
+
+	-o			Specify the name of the output zip archive.
+	-n			Specify the namespace passed to kubectl commands. If not specified the 'default' namespace is used.
+	-cfg		Include Sourcegraph configuration json. Defaults to true.
+
+Examples:
+
+    $ src debug kube -o debug.zip
+
+	$ src -v debug kube -n ns-sourcegraph -o foo
+
+	$ src debug kube -cfg=false -o bar.zip
+
+`
+
+	flagSet := flag.NewFlagSet("kube", flag.ExitOnError)
+	var base string
+	var namespace string
+	var configs bool
+	flagSet.BoolVar(&configs, "cfg", true, "If true include Sourcegraph configuration files. Default value true.")
+	flagSet.StringVar(&namespace, "n", "default", "The namespace passed to kubectl commands, if not specified the 'default' namespace is used")
+	flagSet.StringVar(&base, "o", "debug.zip", "The name of the output zip archive")
+
+	handler := func(args []string) error {
+		if err := flagSet.Parse(args); err != nil {
+			return err
+		}
+
+		//validate out flag
+		if base == "" {
+			return fmt.Errorf("empty -o flag")
+		}
+		// declare basedir for archive file structure
+		var baseDir string
+		if !strings.HasSuffix(base, ".zip") {
+			baseDir = base
+			base = base + ".zip"
+		} else {
+			baseDir = strings.TrimSuffix(base, ".zip")
+		}
```

Comment:
This could be abstracted pretty easily, and seems like it is a place that might be extended later as well (so abstracting now will cause you to change one place instead of two or three).

Do you want to, for example, check that the target file doesn't already exist so you don't overwrite it? Do you want to ensure the directory exists? Create it if it doesn't?

Reply:
I've abstracted this but I'm not sure its what you had in mind. 

> check that the target file doesn't already exist so you don't overwrite it?

I think that zipWriter might be handling this

I tested and it errors out, now with a bad error message -- still figuring out the error formating
```
warrengifford@Warrens-MacBook-Pro src-debug-test % ls | grep bk
bk
bk.zip
warrengifford@Warrens-MacBook-Pro src-debug-test % src debug kube -n ns-sourcegraph -o bk
Archiving kubectl data for 33 pods
 SRC_ENDPOINT: https://cse-aws-test.sgdev.org
 Context: gke_beatrix-test-overlay_us-central1-c_beatrix-test
 Namespace: ns-sourcegraph
 Output filename: bk.zip
Do you want to start writing to an archive? [y/N]: y
failed to write archives from channel: failed to create bk/kubectl/pods/cadvisor-44nvf/cadvisor.log: %!w(*errors.errorString=&{invalid argument}): invalid argument (exit code: 1)

```

Reply:
```
warrengifford@Warrens-MacBook-Pro src-debug-test % src debug server -c 489fc214a746 -o serv
failed to open file: %!w(*fs.PathError=&{open serv.zip 17}): open serv.zip: file exists
```

`src debug server` is failing this better